Source/AliveLib/Scrab.cpp

```diff
@@ -627,8 +685,432 @@ __int16 Scrab::AI_Patrol_0_4AA630()
 
 __int16 Scrab::AI_ChasingEnemy_1_4A6470()
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    Scrab *pScrabToFight = FindScrabToFight_4A4E20();
+    if (pScrabToFight)
+    {
+        SetBrain(&Scrab::AI_Fighting_2_4A5840);
+        field_124_fight_target_obj_id = pScrabToFight->field_8_object_id;
+        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+        return AI_Fighting::eState2_LookingForOpponent_0;
+    }
+    auto pObj = static_cast<BaseAliveGameObject*>(sObjectIds_5C1B70.Find_449CF0(field_120_obj_id));
+    if (!pObj || field_6_flags.Get(BaseGameObject::eDead) || static_cast<int>(sGnFrame_5C1B84) > field_14C && !CanSeeAbe_4A51A0(pObj))
+    {
+        field_120_obj_id = -1;
+        field_108_next_motion = eScrabMotions::M_Stand_0_4A8220;
+        ToPatrol_4AA600();
+        return AI_Patrol::eState0_ToMoving_0;
+    }
+    if (pObj->field_114_flags.Get(Flags_114::e114_Bit8_bInvisible))
+    {
+        field_120_obj_id = -1;
+        field_108_next_motion = eScrabMotions::M_HowlBegin_26_4A9DA0;
+        ToPatrol_4AA600();
+        return AI_Patrol::eState0_UNKNOWN_9;
+    }
+    if (CanSeeAbe_4A51A0(pObj))
+    {
+        field_14C = sGnFrame_5C1B84 + field_148_attack_duration;
+    }
+    if (Event_Is_Event_In_Range_422C30(kEventAbeOhm, field_B8_xpos, field_BC_ypos, -1) && field_11C_sub_state != 26)
+    {
+        field_108_next_motion = eScrabMotions::M_HowlBegin_26_4A9DA0;
+        return AI_ChasingEnemy::eState1_Panic_4;
+    }
+    LiftPoint* pLiftPoint = static_cast<LiftPoint*>(pObj);
+    if (!pObj || pObj->field_4_typeId != Types::eLiftPoint_78 || pLiftPoint->vOnAnyFloor_461920() || field_11C_sub_state == 4)

```

Comment:
I think you should use pLiftPoint instead of pObj in the if statement as the are actually the same thing

Reply:
I think if you invert this whole if statement you can remove the else block first with an early return